File: core/src/main/java/glitch/auth/objects/json/AccessToken.java
package glitch.auth.objects.json;

import com.google.gson.annotations.SerializedName;
import glitch.auth.GlitchScope;

import javax.annotation.Nullable;
import java.time.Instant;
import java.util.Set;

public interface AccessToken {
    String getAccessToken();

    String getRefreshToken();

    @Nullable
    @SerializedName("expires_in")
    Instant getExpiredAt();

    @SerializedName("scope")
    Set<GlitchScope> getScopes();
}

File: core/src/main/kotlin/glitch/auth/objects/json/AccessToken.kt
package glitch.auth.objects.json

import com.google.gson.annotations.SerializedName
import glitch.api.objects.json.interfaces.Creation
import glitch.auth.GlitchScope

interface AccessToken : Creation {
    val accessToken: String

    val refreshToken: String

    @get:SerializedName("scope")
    val scopes: Set<GlitchScope>
}
